pkg/manager/manager.go

```diff
@@ -204,6 +212,15 @@ type Options struct {
 	// between tries of actions. Default is 2 seconds.
 	RetryPeriod *time.Duration
 
+	// OnStoppedLeadingCallback provides a way to configure a set of helper functions
+	// that are invoked when the current leader stops leading by losing the election or failing
+	// to renew the lock.
+	OnStoppedLeadingCallback []func() error
```

Comment:
Do we really need the function list? Maybe one callback for users can hold multiple code block or functions calling in it.

Reply:
Keeping them as an array so that the registration for this callback can be done in more than one place. It is not always easy to aggregate all the callbacks in one uber callback from UT perspective right ? So, the provision to register multiple smaller callbacks programatically from different places in the code base before the `Manager` instance can be created.